src/github_proxy.py

```diff
@@ -46,14 +54,29 @@ def publish_pr_comment(self, build):
             logs_url=build.get_logs_url(),
         )
 
-        # initialize client before logging to ensure GitHub attributes are populated
-        gh_client = self._get_client()
+        repo = self._get_repo()
         LOG.debug('Publishing PR Comment: repo=%s/%s, pr_id=%s, comment=%s',
                   self._github_owner, self._github_repo, build.get_pr_id(), pr_comment)
-
-        repo = gh_client.get_user(self._github_owner).get_repo(self._github_repo)
         repo.get_pull(build.get_pr_id()).create_issue_comment(pr_comment)
 
+    def delete_previous_comments(self, build):
+        """Delete previous PR comments."""
+        repo = self._get_repo()
+        for comment in repo.get_issue(build.get_pr_id()).get_comments():
+            if HIDDEN_COMMENT in comment.body:  # Check for hidden comment in body
+                try:  # Not critical, catch all GitHub exceptions here
+                    LOG.debug('Deleting previous comment: repo=%s/%s, pr_id=%s, comment_id=%s',
+                              self._github_owner, self._github_repo, build.get_pr_id(), comment.id)
+                    comment.delete()
+                except GithubException:
+                    pass
```

Comment:
Can you add a LOG.warning here instead of just silently succeeding? Picturing something like this:

```python
                except GithubException as e:
                    LOG.warning('Failed to delete previous comment: repo=%s/%s, pr_id=%s, comment_id=%s, error=%s',
                                self._github_owner, self._github_repo, build.get_pr_id(), comment.id, str(e))
```